hash.rb

```diff
@@ -1,39 +1,35 @@
 class Hash
   # call-seq:
-  #    Hash.new(default_value = nil) -> new_hash
-  #    Hash.new(default_value = nil, capacity: size) -> new_hash
-  #    Hash.new {|hash, key| ... } -> new_hash
-  #    Hash.new(capacity: size) {|hash, key| ... } -> new_hash
-  #
-  # Returns a new empty +Hash+ object.
-  #
-  # The initial default value and initial default proc for the new hash
-  # depend on which form above was used. See {Default Values}[rdoc-ref:Hash@Default+Values].
-  #
-  # If neither an argument nor a block is given,
-  # initializes both the default value and the default proc to <tt>nil</tt>:
-  #   h = Hash.new
-  #   h.default # => nil
-  #   h.default_proc # => nil
-  #
-  # If argument <tt>default_value</tt> is given but no block is given,
-  # initializes the default value to the given <tt>default_value</tt>
-  # and the default proc to <tt>nil</tt>:
-  #   h = Hash.new(false)
-  #   h.default # => false
-  #   h.default_proc # => nil
-  #
-  # If a block is given but no <tt>default_value</tt>, stores the block as the default proc
-  # and sets the default value to <tt>nil</tt>:
-  #   h = Hash.new {|hash, key| "Default value for #{key}" }
-  #   h.default # => nil
-  #   h.default_proc.class # => Proc
-  #   h[:nosuch] # => "Default value for nosuch"
-  #
-  # If both a block and a <tt>default_value</tt> are given, raises an +ArgumentError+
-  #
-  # If the optional keyword argument +capacity+ is given, the hash will be allocated
-  # with enough capacity to accommodate this many keys without having to be resized.
+  #   Hash.new(default_value = nil, capacity: 0)
+  #   Hash.new(capacity: 0) {|self, key| ... } -> new_hash
+  #
+  # Returns a new empty \Hash object;
+  # initializes the values of Hash#default and Hash#default_proc,
+  # which determine the value to be returned by method Hash#[] when the entry does not exist.
+  #
+  # With no block given, initializes Hash#default to the given +default_value+,
+  # and Hash#default_proc to +nil+:
+  #
+  #   h = Hash.new        # => {}
+  #   h.default           # => nil
+  #   h.default_proc      # => nil
+  #   h = Hash.new(false) # => {}
+  #   h.default           # => false
+  #
+  # With a block given, initializes Hash#default to +nil+,
+  # and Hash#default_proc to a new Proc containing the block's code:
+  #
+  #   h = Hash.new {|hash, key| "Hash #{hash}: Default value for #{key}" }
+  #   h.default      # => nil
+  #   h.default_proc # => #<Proc:0x00000289b6fa7048 (irb):185>
+  #   h[:nosuch]     # => "Hash {}: Default value for nosuch"
+  #
+  # If optional keyword argument +capacity+ is given with a positive integer value +n+,
+  # initializes the hash with enough capacity to accommodate +n+ keys without resizing.
+  #
+  # Raises ArgumentError if both +default_value+ and a block are given.
+  #
+  # See also [Methods for Creating a Hash](rdoc-ref:Hash@Methods+for+Creating+a+Hash).
```

Comment:
We should use RDoc link syntax here:

```suggestion
  # See also {Methods for Creating a Hash}[rdoc-ref:Hash@Methods+for+Creating+a+Hash].
```